front/lib/api/assistant/conversation.ts

```diff
@@ -1075,6 +1089,36 @@ export async function* postUserMessage(
   void logIfUserUnknown();
 }
 
+/**
+ * Can a user mention a given configuration
+ */
+async function canAccessAgent(
+  auth: Authenticator,
+  agentConfiguration: LightAgentConfigurationType
+): Promise<boolean> {
+  if (auth.isAdmin() || agentConfiguration.status === "draft") {
+    return true;
+  }
+
+  // Global agent
+  if (agentConfiguration.versionAuthorId === null) {
+    return agentConfiguration.status === "active";
+  }
+
+  if (agentConfiguration.scope === "hidden") {
+    if (agentConfiguration.status === "active") {
+      const group = await GroupResource.fetchByAgentConfiguration(
```

Comment:
Mmmmmmh are these groups all in place already? Won't that break existing flows while this is all behind a feature flag? (Just want to make sure here)

Also I think we want  to avoid a double SQL call on this one. We likely will be pulling this information somewhere else at some point and would rather have it provided to this function? Do you know what's the plan here?

cc @tdraier

Reply:
1. Shouldn't break current flow, but won't be active for it as indeed it target `hidden` scope. So for current agent it would not go into that section.

2. Not sure if we'll pull more info, I'll ask @tdraier. 
Agree we could optimize the query. I would go for a static `isMemberOf(agentConfigurationIds: string[]): Promise<Record<configId, boolean>>`, that fetch the need info in one go, we could call it in the current [`Promise.all`](https://github.com/dust-tt/dust/pull/12425/files#diff-8eaf52f21ee24773c15978a8654aad5ab5041354101dfbf87e6e5112293d0753L1190)

Reply:
Just want to make sure we don't have this one hidden here and add another one in 2 days a layer above. So the main question is where is that GroupResource call going to be made to be fully optimized and try to get the final state in that PR to avoid the double SQL query down the line :+1:

Reply:
Totally, I'll wait for some feedback from @tdraier to improve that fix

Reply:
:ok_hand: